### ISSUE TITLE
Format number with K/Ms

### ISSUE THREAD

**Issue by cmutagorama** [thijssmudde/react-headless-pagination#9]
I have a large dataset to display but some numbers are overlapping due to their long length.
<img width="1098" alt="Screenshot 2023-03-15 at 14 21 06" src="https://user-images.githubusercontent.com/36790815/225307712-b561261c-3589-489b-9548-efd1013be1e3.png">

Can I have a way to format number so that instead of displaying 40000 => 40k and so on?

**Comment by cmutagorama** [comment]
@thijssmudde any update on this?

**Comment by thijssmudde** [comment]
Yes, it's possible to add this functionality as a render prop. However, I'm not sure yet if this library should support this behaviour.

**Comment by cmutagorama** [comment]
Not supporting this feature/behavior would mean that this lib is limited to a certain number of total pages, right?
